tendrl/node_agent/node_sync/sds_detect.py

```diff
@@ -37,49 +38,64 @@ def sync():
+                            if dc.detected_cluster_id:
+                                if dc.detected_cluster_id != sds_details.get(
+                                    'detected_cluster_id'):
+                                    # Gluster peer list has changed
+                                    integration_id = \
+                                        NS.tendrl_context.integration_id
+                                    etcd_utils.write(
+                                        integration_index_key,
+                                        integration_id
+                                    )
+                                    params = {
+                                        'TendrlContext.integration_id': integration_id,
+                                    }
+                                    payload = {
+                                        "tags": ["provisioner/%s" % integration_id],
+                                        "run": "tendrl.flows.ExpandClusterWithDetectedPeers",
+                                        "status": "new",
+                                        "parameters": params,
+                                        "type": "node"
+                                    }
+                                    _job_id = str(uuid.uuid4())
+                                    Job(job_id=_job_id, status="new",
+                                        payload=payload).save()
                             else:
                                 integration_id = str(uuid.uuid4())
-                                NS._int.wclient.write(
-                                    integration_index_key,
-                                    integration_id,
-                                    prevExist=False
+                                etcd_utils.write(integration_index_key,
+                                                 integration_id
                                 )
-                        except etcd.EtcdAlreadyExist:
-                            if not dc_changed:
-                                integration_id = NS._int.client.read(
-                                    integration_index_key).value
-                        finally:
-                            NS.tendrl_context.integration_id = integration_id
-                            NS.tendrl_context.cluster_id = sds_details.get(
-                                'detected_cluster_id')
-                            NS.tendrl_context.cluster_name = sds_details.get(
-                                'detected_cluster_name')
-                            NS.tendrl_context.sds_name = sds_details.get(
-                                'pkg_name')
-                            NS.tendrl_context.sds_version = sds_details.get(
-                                'pkg_version')
-                            NS.tendrl_context.save()
+
+                                
+                        while True:
```

Comment:
How do we make sure the above job is completed or failed? Should we wait indefinitely here?

Reply:
Yes, the result of this job is crucial for the cluster, so we should wait indefinitely